src/DefinitionResolver.php

```diff
@@ -896,11 +896,12 @@ public function resolveClassNameToType($class): Type
             // Anonymous class
             return new Types\Object_;
         }
-        $className = (string)$class->getResolvedName();
-
-        if ($className === 'static') {
+        if ($class instanceof PhpParser\Token && $class->kind === PhpParser\TokenKind::StaticKeyword) {
+            // `new static`
             return new Types\Static_;
         }
+        $className = (string)$class->getResolvedName();
+
         if ($className === 'self' || $className === 'parent') {
```

Comment:
There aren't TokenKinds for self and parent. I don't know why. They are parsed as QualifiedNames.